compiler/il/OMRBlock.cpp

```diff
@@ -805,6 +807,687 @@ OMR::Block::splitWithGivenMethodSymbol(TR::ResolvedMethodSymbol *methodSymbol, T
+                     pinningArray->setPinningArrayPointer();
+                     }
+                  }
+               else
+                  {
+                  TR::SymbolReference *newValueArrayRef = comp->getSymRefTab()->
+                                                         createTemporary(methodSymbol, TR::Address, false, 0);
+                  TR::Node *newStore = TR::Node::createStore(newValueArrayRef, valueChild);
+                  TR::TreeTop *newStoreTree = TR::TreeTop::create(comp, newStore);
+                  insertBefore->insertBefore(newStoreTree);
+                  if (!newValueArrayRef->getSymbol()->isParm()) // newValueArrayRef could contain a parm symbol
+                     {
+                     pinningArray = newValueArrayRef->getSymbol()->castToAutoSymbol();
+                     pinningArray->setPinningArrayPointer();
+                     }
+                  }
+               }
+            }
+         else
+            {
+            pinningArray = value->getSymbolReference()->getSymbol()->castToInternalPointerAutoSymbol()->getPinningArrayPointer();
+            }
+         symRef->getSymbol()->castToInternalPointerAutoSymbol()->setPinningArrayPointer(pinningArray);
+         if (value->isInternalPointer() && pinningArray)
+            {
+            value->setPinningArrayPointer(pinningArray);
+            } 
+         }
+      }
+   
+   if (dataType == TR::Aggregate)
+      {
+      uint32_t size = value->getSize();
+      symRef = new (comp->trHeapMemory()) TR::SymbolReference(comp->getSymRefTab(),
+                         TR::AutomaticSymbol::create(comp->trHeapMemory(),dataType,size),
+                         methodSymbol->getResolvedMethodIndex(),
+                         methodSymbol->incTempIndex(comp->fe()));
+
+
+      if (value->isNotCollected())
+         symRef->getSymbol()->setNotCollected();
+      }
+
+   if (!symRef)
+      {
+      symRef = comp->getSymRefTab()->createTemporary(methodSymbol, value->getDataType());
+      if (value->getType() == TR::Address && value->isNotCollected())
+         symRef->getSymbol()->setNotCollected();
+      }
+   return symRef;
+   }
+
+/**
+ * Splits the blocks from the split point manually handling the GlRegDeps created by the Global Register Allocator.
+ * If there are available global registers which can be used, it will utililze them to uncommon nodes across split points.
+ * 
+ * @param startOfNewBlock TreeTop from which new block will start
+ * @param cfg TR::CFG object
+ * @param copyExceptionSuccessors Boolean stating if we need to copy the exceptionSuccessors of the original blocks to new block
+ * @param methodSymbol TR::ResolvedMethodSymbol object
+ * @param trace Boolean stating if tracing is requested
+ * @return TR::Block* Returns new Block
+ */
+TR::Block *
+OMR::Block::splitPostGRA(TR::TreeTop * startOfNewBlock, TR::CFG *cfg, bool copyExceptionSuccessors, TR::ResolvedMethodSymbol *methodSymbol, bool trace)
```

Comment:
Do we have precedence for passing the `trace` boolean as an argument? Looking at related code it seems we initialize the trace boolean in the function itself. IMO this complicates the API unnecessarily.

Reply:
Those traces were used for debug purpose only to make sure API is functionally correct and to control amount of text we have in log file. I have removed it to simplify the API in https://github.com/eclipse/omr/pull/4152/commits/e75062fc5ab9ebb3996cc60947cb12fb2f027537